### ISSUE TITLE
New Rule: A rule that disallows `no-unnecessary-await`s

### ISSUE THREAD

**Issue by markcellus** [eslint/eslint#16959]
### Rule details

I'd like to request that a rule be added that will error when an unnecessary `await` call exists to a synchronous function. [`await-thenable`](https://typescript-eslint.io/rules/await-thenable/) exists for this purpose, but I'd like to avoid using Typescript just for this behavior.

### Related ECMAScript feature

async/await

### What type of rule is this?

Suggests an alternate way of doing something

### Example code

```js
await 'value';
// OR
await someSyncFunction();

```

### Why should this rule be in the core instead of a plugin?

It can be in a plugin I guess, but I don't know of any that exists for this. async/await is a core feature in ECMAScript, so core seems like the optimal place for it.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

**Comment by mdjermanovic** [comment]
Hi @markcellus, thanks for the issue!

> I'd like to request that a rule be added that will error when an unnecessary `await` call exists to a synchronous function.

Core rules don't have type information to know that a function doesn't return a Promise, so this rule doesn't seem feasible for core.
